core/src/main/java/tc/oc/pgm/flag/state/Carried.java

```diff
@@ -215,29 +215,39 @@ public void tickRunning() {
 
   @Override
   public boolean isCarrying(MatchPlayer player) {
-    return this.carrier == player;
+    if (this.pickingUp == null) {
+      return this.carrier == player;
+    } else {
+      return this.pickingUp == player;
+    }
   }
 
   @Override
   public boolean isCarrying(Party party) {
-    return this.carrier.getParty() == party;
+    if (this.pickingUp == null) {
+      return this.carrier.getParty() == party;
+    } else {
+      return this.pickingUp.getParty() == party;
+    }
   }
 
   @Override
   protected boolean canSeeParticles(Player player) {
     return player != this.carrier.getBukkit();
   }
 
-  protected void dropFlag() {
+  @Override
+  public boolean dropFlag() {
     for (Location dropLocation : this.dropLocations) {
       if (this.flag.canDrop(new PlayerQuery(null, carrier, dropLocation))) {
         this.flag.transition(new Dropped(this.flag, this.post, dropLocation, this.carrier));
-        return;
+        return true;
```

Comment:
Is the return value used for anything? As far as i can see the result is just ignored from the caller